### PR TITLE
Parquet: Ignore INT96 metrics while importing tables

### PR DESCRIPTION
**Pull request by aokolnychyi**
We fail while importing existing Parquet tables with INT96 timestamps as `ParquetConversions.fromParquetPrimitive` assumes the value is long. Since INT96 metrics are unreliable, I propose to ignore them.

```
java.lang.ClassCastException: org.apache.iceberg.shaded.org.apache.parquet.io.api.Binary$ByteArrayBackedBinary cannot be cast to java.lang.Long
	at org.apache.iceberg.parquet.ParquetConversions.fromParquetPrimitive(ParquetConversions.java:48)
	at org.apache.iceberg.parquet.ParquetUtil.footerMetrics(ParquetUtil.java:133)
	at org.apache.iceberg.spark.SparkTableUtil.lambda$listParquetPartition$8(SparkTableUtil.java:426)
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175)
	at java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
```

### CODE REVIEW

**Comment by aokolnychyi** [comment]
@rdblue @thesquelched @gustavoatt could you take a look, please?

File: parquet/src/main/java/org/apache/iceberg/parquet/ParquetUtil.java [hunk 1]
```diff
@@ -103,6 +103,11 @@ public static Metrics footerMetrics(ParquetMetadata metadata, MetricsConfig metr
     for (BlockMetaData block : blocks) {
       rowCount += block.getRowCount();
       for (ColumnChunkMetaData column : block.getColumns()) {
+        if (column.getPrimitiveType().getPrimitiveTypeName() == PrimitiveType.PrimitiveTypeName.INT96) {
+          // skip metrics for INT96 timestamps
```
Comment (rdblue) [inline parquet/src/main/java/org/apache/iceberg/parquet/ParquetUtil.java:107]: We may want to include a justification for skipping: the stats are not reliable.

Comment (aokolnychyi) [reply parquet/src/main/java/org/apache/iceberg/parquet/ParquetUtil.java:107]: Will do

Comment (aokolnychyi) [reply parquet/src/main/java/org/apache/iceberg/parquet/ParquetUtil.java]: Done

**Comment by aokolnychyi** [comment]
I've updated the PR to include size and null values counts but not min/max values.

**Comment by aokolnychyi** [comment]
Wait, I don't handle nesting correctly.

**Comment by rdblue** [comment]
Thanks for fixing this, @aokolnychyi! Merging it.

**Pull request merged**